cdap-metrics-sidecar/src/test/java/io/cdap/cdap/jmx/metrics/JMXMetricsCollectorTest.java

```diff
@@ -0,0 +1,144 @@
+import io.cdap.cdap.api.metrics.MetricsCollectionService;
+import io.cdap.cdap.api.metrics.MetricsContext;
+import io.cdap.cdap.common.conf.CConfiguration;
+import io.cdap.cdap.common.conf.Constants;
+import io.cdap.cdap.common.utils.Tasks;
+import io.cdap.cdap.proto.id.NamespaceId;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.mockito.exceptions.verification.TooLittleActualInvocations;
+
+import java.io.IOException;
+import java.lang.management.ManagementFactory;
+import java.net.MalformedURLException;
+import java.rmi.registry.LocateRegistry;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import javax.management.MBeanServer;
+import javax.management.remote.JMXConnectorServer;
+import javax.management.remote.JMXConnectorServerFactory;
+import javax.management.remote.JMXServiceURL;
+
+import static org.mockito.Matchers.anyLong;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.atLeast;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class JMXMetricsCollectorTest {
+  private static final int SERVER_PORT = 11023;
+  private static final String SERVICE_NAME = "test-service";
+  private static JMXConnectorServer svr;
+
+  private static final Map<String, String> METRICS_CONTEXT = ImmutableMap.of(
+    Constants.Metrics.Tag.NAMESPACE, NamespaceId.SYSTEM.getNamespace(),
+    Constants.Metrics.Tag.COMPONENT, SERVICE_NAME);
+
+  @Mock
+  private MetricsCollectionService mockMetricsService;
+  @Mock
+  private MetricsContext mockContext;
+  @Mock
+  private Environment mockEnv;
+
+  @Before
+  public void beforeEach() {
+    MockitoAnnotations.initMocks(this);
+    when(mockMetricsService.getContext(METRICS_CONTEXT)).thenReturn(mockContext);
+  }
+
+  @BeforeClass
+  public static void setupClass() throws IOException {
+    svr = createJMXConnectorServer(SERVER_PORT);
+    svr.start();
+    Assert.assertEquals(svr.isActive(), true);
+  }
+
+  @AfterClass
+  public static void teardownClass() throws IOException {
+    svr.stop();
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void testInvalidPortInConfig() throws Exception {
+    when(mockEnv.getVariable(Constants.JMXMetricsCollector.COMPONENT_NAME_ENV_VAR)).thenReturn(SERVICE_NAME);
+    CConfiguration cConf = CConfiguration.create();
+    cConf.setInt(Constants.JMXMetricsCollector.SERVER_PORT, -1);
+    cConf.setInt(Constants.JMXMetricsCollector.POLL_INTERVAL, 100);
+    JMXMetricsCollector jmxMetrics = new JMXMetricsCollector(cConf, mockMetricsService, mockEnv);
+    jmxMetrics.start();
+    jmxMetrics.stop();
+    throw new Exception("Service should not have started successfully with Invalid Port");
```

Comment:
The test will fail f there is no IllegalArgumentException, so throwing exception here is not required.

Reply:
Got it, removed the exception.